### ISSUE TITLE
WebDateField fixed width ?

### ISSUE THREAD

**Issue by mokun** [mgarin/weblaf#580]
Hi, 

How do I reduce the width of the WebDateField ?

It doesn't adjust dynamically.

Currently it's too wide for my app. 

See below the extra whitespace that I circle with a red oval : 

![webdatefield](https://user-images.githubusercontent.com/1168584/68341909-5343b380-009e-11ea-9cb9-51724f68997d.jpg)

How do I manually adjust it ?

I've tried `setSize(100, 30):` but it can't change the width.

**Comment by mgarin** [comment]
Resulting size of the field is either determined by it's preferred size (which usually makes sure that all information in the component is visible - text in the field in this case) or by it's container's layout.

Looking at the screenshot - I'd guess that layout decides the field size in you case, so changing it's size will have no effect because size (and bounds which include size & location) only apply to containers with `null` layout. 

Changing field's preferred size might have effect, but it will depend on how your field's container layout distributes remaining space between it's components. In this particular I doubt it would have any effect either (except for maybe height of the field) because field is stretched by the layout already, meaning it ignores it's current smaller preferred size.

There might also be some issues with field size calculation in theory, but I'm not seeing that happen on any demos I have, so I can't really help any further without knowing the code of the field and container you're using.

**Comment by mokun** [comment]
Yes. The preferred size works. I use `setPreferredWidth()` since I don't need to override its heigfht. Thanks !

Q: how do I override the font and color of the textfield inside ?

The setForeground() and setFont() below don't work : 

```
Date date = new Date(earthClock.getInstant().toEpochMilli()));
WebDateField dateField = new WebDateField(date);
dateField.setPreferredWidth(190);
dateField.setFont(font);
dateField.setForeground(Color.BLUE);
```

**Comment by mgarin** [comment]
> Q: how do I override the font and color of the textfield inside ?

It's not possible in the latest released version, but I've added a small improvement that will forward `Font` and foreground `Color` provided into `WebDateField` to the field. It will soon be available in snapshot build of v1.2.11.

---

Also, regarding the field length - I actually know why it becomes so lengthy now. It adjusts it's columns according to your date pattern. It's not the best solution because it makes fields really long in some cases. 

I've added some internal improvements for that, you might not need the custom preferred width anymore on newer version, for instance this is the new preferred size for your pattern:

![image](https://user-images.githubusercontent.com/2361389/68466110-ca0ab900-0224-11ea-828b-1888b42fc134.png)

A few extra pixels are added to ensure text will fit in most cases, otherwise it stays really close to actual text width now.
